PRCS/prc-2.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Paima Hololocker Interface
+description: Interface for projecting ERC721 tokens on EVM networks for usage in Paima
+author: matejos (@matejos)
+status: Draft
+created: 2023-11-08
+---
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for projecting ERC721 tokens on EVM networks for usage in Paima applications. This standard provides basic functionality to lock tokens, request to unlock locked tokens, and withdraw tokens after certain time has passed since the unlock request.
+
+## Motivation
+
+Many games, due to being data and computation heavy applications, run on sidechains, L2s and appchains as opposed to popular L1 blockchains. This is problematic because popular NFT collections (which people generally want to use in-game) live on the L1 (a different environment). A common solution to this problem is building an NFT bridge, but bridges not only have a bad reputation for fungible tokens which limits usage, the problem is even worse for NFTs where there is also a philosophical disconnect (if a bridge gets hacked, which is the canonical NFT? The one the hacker stole, or the bridged asset?)
+
+Instead of bridging NFTs, this standard encourages users to project their NFT directly into the game, allowing them to access their asset in-game without having to bridge it to the game chain. Although the main use-case is projecting a single NFT, it supports projecting multiple NFTs at once as well.
+
+## Specification
+
+Every PRC-2 compliant contract must implement the `HololockerInterface` interface:
+
+```solidity
+interface HololockerInterface is IERC721Receiver {
+    /// @dev Data structure that must exist for each locked NFT
+    struct LockInfo {
+        /// Timestamp when NFT will be withdrawable, 0 if unlock hasn't been requested
+        uint256 unlockTime;
+        /// Rightful owner of the NFT
+        address owner;
+        /// Account that initiated the lock
+        address operator;
+    }
+
+    /// @dev This emits when NFT is locked, either via lock function or via NFT being sent to this contract
+    /// @param token NFT address
+    /// @param owner Rightful owner of the NFT
+    /// @param tokenId NFT token identifier
+    /// @param operator Address initiating the lock
+    event Lock(address indexed token, address indexed owner, uint256 tokenId, address operator);

```

Comment:
Maybe we should split the specification into a "common" part and an ERC721-specific part

that way for other standards like erc20/erc1155 we can add a header for the parts that changed instead on top of the common base

Reply:
The only common part is the `event LockTimeUpdate`, `function getLockTime`, and `function setLockTime`, so I'm not sure if it brings value to split it 🤔